### PR TITLE
 customresourcevalidation: switch Features to FeatureGate 

### PR DESCRIPTION
**Pull request by sjenning**
@adambkaplan @rphillips @derekwaynecarr 

addresses https://github.com/openshift/api/pull/247#issuecomment-474040956

### CODE REVIEW

File: pkg/admission/customresourcevalidation/features/validate_features.go [hunk 1]
```diff
@@ -26,22 +26,22 @@ func Register(plugins *admission.Plugins) {
...
 			},
 			map[schema.GroupVersionKind]customresourcevalidation.ObjectValidator{
-				configv1.GroupVersion.WithKind("Feature"): featuresV1{},
+				configv1.GroupVersion.WithKind("FeatureGate"): featuresV1{},
 			})
 	})
 }
 
-func toFeatureV1(uncastObj runtime.Object) (*configv1.Features, field.ErrorList) {
+func toFeatureV1(uncastObj runtime.Object) (*configv1.FeatureGate, field.ErrorList) {
```
Comment (adambkaplan) [inline pkg/admission/customresourcevalidation/features/validate_features.go]: nit: /s/toFeatureV1/toFeatureGateV1/

File: pkg/admission/customresourcevalidation/features/validate_features.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ type featuresV1 struct {
 
 var knownFeatureSets = sets.NewString("", string(configv1.TechPreviewNoUpgrade))
 
-func validateFeatureSpecCreate(spec configv1.FeaturesSpec) field.ErrorList {
+func validateFeatureSpecCreate(spec configv1.FeatureGateSpec) field.ErrorList {
```
Comment (adambkaplan) [inline pkg/admission/customresourcevalidation/features/validate_features.go]: nit: rename to `validateFeatureGateSpecCreate`

File: pkg/admission/customresourcevalidation/features/validate_features.go [hunk 3]
```diff
@@ -64,7 +64,7 @@ func validateFeatureSpecCreate(spec configv1.FeaturesSpec) field.ErrorList {
 	return allErrs
 }
 
-func validateFeatureSpecUpdate(spec, oldSpec configv1.FeaturesSpec) field.ErrorList {
+func validateFeatureSpecUpdate(spec, oldSpec configv1.FeatureGateSpec) field.ErrorList {
```
Comment (adambkaplan) [inline pkg/admission/customresourcevalidation/features/validate_features.go]: nit: rename to `validateFeatureGateSpecUpdate`

**Comment by adambkaplan** [comment]
@sjenning I also had to run `make update` to pull in changes to openapi

**Comment by sjenning** [comment]
@adambkaplan ok I think I got everything

**Pull request merged**